packages/flutter_tools/lib/src/build_system/targets/linux.dart

```diff
@@ -124,6 +125,10 @@ abstract class BundleLinuxAssets extends Target {
       environment.buildDir.childFile('app.dill')
         .copySync(outputDirectory.childFile('kernel_blob.bin').path);
     }
+    final String versionInfo = FlutterProject.current().getVersionInfo();
+    outputDirectory
```

Comment:
Instead of writing this directly to the output directory, you should use the `additionalContent` field in copyAssets to ensure this is added to the asset manifest.